src/stan/callbacks/concurrent_writer.hpp

```diff
@@ -0,0 +1,168 @@
+#ifndef STAN_CALLBACKS_CONCURRENT_WRITER_HPP
+#define STAN_CALLBACKS_CONCURRENT_WRITER_HPP
+
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/prim/meta.hpp>
+#include <tbb/concurrent_queue.h>
+#include <condition_variable>
+#include <functional>
+#include <string>
+#include <thread>
+#include <vector>
+
+namespace stan::callbacks {
+#ifdef STAN_THREADS
+/**
+ * Takes a writer and makes it thread safe via multiple queues.
```

Comment:
optional:
```suggestion
 * Enables thread-safe writing of numeric values to a writer
```
I think this comment is when the class was trying to be more general